### ISSUE TITLE
How can I Get, Edit and Set Gradient Matrix in Training of Keras Model ?

### ISSUE THREAD

**Issue by HumzaSami00** [keras-team/keras#16569]
**System information**.
TensorFlow version 2.5
---------------------------------
I am creating a sparse neural network as described in the image below. Keras only provides a dense layer and we can't choose how many neurons we want to be connected to the previous layer. For implementing this using Keras, I am trying to implement the following approach:

1- Get the Gradient Matrix of each layer in each epoch
2- Multiply Gradient Matrix with a mask matrix to make it sparse 
3- Update new weights according to that updated gradient matrix

I could not be able to find a gradient matrix in Keras. How can I get it and update it during epochs? 
Thanks in Advance
![Image from iOS (1)](https://user-images.githubusercontent.com/101699223/169794653-6b55902b-838a-4886-8b32-3650d0b1d168.png)

**Comment by tilakrayal** [comment]
Hi @HumzaSami00,
 In order to expedite the trouble-shooting process, could you provide the complete code to reproduce the issue reported here. Thank you!
